### PR TITLE
Feature/allow bolded hyperlinks

### PR DESCRIPTION
**Pull request by dragon-dxw**
The `Contentful-converter` gem swallows links inside bold tags.
Why is hyperlinks-in-bold a problem? There's a hierarchy in [contentful's rich text format](https://www.contentful.com/developers/docs/concepts/rich-text/#rules-of-rich-text ): blocks (some of which can contain blocks, recursively), inline elements and text.
Bold exists within text, but hyperlinks are an inline element. So rather than saying:
`<b>lion, <a href='panther.txt'>ocelot</a>, puma<b>`
we need to keep everything ordered for the same effect:
`<b>lion, </b><a href='panther.txt'><b>ocelot</b></a><b>, puma</b>`

### CODE REVIEW

File: spec/lib/mangle_xml_spec.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+
+RSpec.describe "When the mangler recieves XML" do
+  it "puts the bold tags into the hyperlinks" do
```
Comment (tahb) [inline spec/lib/mangle_xml_spec.rb]: I understand the concept of the earlier comment that says:

> `<b><a>.</a></b> => <a><b>.</b></a>`

I see that tags `<b>`, `<i>` and `<u>` all seem to be covered by the magic mangling logic 👍 

In the tests though this seems to be covering b and u to some extent? Should we have a test for each and a test with no links to make sure it doesn't make changes?

File: spec/lib/mangle_xml_spec.rb [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
-  it "puts the bold tags into the hyperlinks" do
-    html = CAEWCorporateImporter::Mangler.new.mangle_xml("<p>Did you know? <b>this <u>is a <a href='kitten.txt'>kitten</a>!</u></b></p>")
-    expect(html).to include('<body><p>Did you know? <b>this <u>is a </u><a href="kitten.txt"><u>kitten</u></a><u>!</u></b></p></body>')
+RSpec.describe "When the recursive mangler recieves XML" do
+  it "puts the bold and underline tags into the hyperlinks" do
+    html = CAEWCorporateImporter::Mangler.new.recursive_mangle(
+      "<p>Did you know? <b>this <u>is a <a href='kitten.txt'>kitten</a>!</u></b></p>"
+    )
+    expect(html).to include(
+      '<p>Did you know? <b>this </b><u><b>is a </b></u><a href="kitten.txt"><b><u>kitten</u></b></a><u><b>!</b></u></p>'
```
Comment (tahb) [inline spec/lib/mangle_xml_spec.rb:9]: This test is much clearer to me 👍

**Review by tahb — APPROVED** [review]
Looks good. Lots of impressive magic happening. Happy we have a test, suggestions for more tests if they are quick to do to increase the level of confidence.

File: spec/lib/mangle_xml_spec.rb (hunk 1 above)
Comment (dragon-dxw) [reply spec/lib/mangle_xml_spec.rb]: I'm pretty happy that if the logic for two tags we're treating symmetrically are OK, the logic is OK; but catching the case where we've failed to be symmetrical isn't necessarily a bad thing.

No link test sounds like a good plan.

**Pull request merged**